File: musheng-tax-system/musheng-config/src/main/java/com/musheng/config/marketplace/entity/Marketplace.java
package com.musheng.config.marketplace.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.musheng.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 站点实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_marketplace")
public class Marketplace extends BaseEntity {

    /**
     * 站点编码(US/CA/UK/DE)
     */
    private String siteCode;

    /**
     * 站点名称
     */
    private String siteName;

    /**
     * Marketplace标识(amazon.com等)
     */
    private String marketplaceId;

    /**
     * 关联货币编码
     */
    private String currencyCode;

    /**
     * 卖家ID
     */
    private String sellerId;

    /**
     * 表头语言(EN/DE)
     */
    private String headerLanguage;

    /**
     * 日期解析格式
     */
    private String dateFormat;

    /**
     * 数字格式(.或,)
     */
    private String numberFormat;

    /**
     * 时区
     */
    private String timezone;

    /**
     * 状态(1启用, 0禁用)
     */
    private Integer status;
}

File: musheng-tax-web/src/types/marketplace.ts
/**
 * 站点（Marketplace）管理类型定义
 * 与后端 t_marketplace 表结构对齐
 */

// 站点信息
export interface Marketplace {
  id: number
  siteCode: string
  siteName: string
  marketplaceId?: string
  currencyCode: string
  sellerId?: string
  headerLanguage?: string
  dateFormat?: string
  numberFormat?: string
  timezone?: string
  status: number
  createTime: string
  updateTime?: string
}

// 站点查询参数
export interface MarketplaceQuery {
  siteCode?: string
  siteName?: string
  status?: number
  page?: number
  size?: number
}

// 站点创建/更新参数
export interface MarketplaceForm {
  id?: number
  siteCode: string
  siteName: string
  marketplaceId?: string
  currencyCode: string
  sellerId?: string
  headerLanguage?: string
  dateFormat?: string
  numberFormat?: string
  timezone?: string
  status?: number
}
